modules/nested-accordion/assets/js/frontend/handlers/nested-accordion.js

```diff
@@ -34,4 +50,123 @@ export default class NestedAccordion extends Base {
 			$items[ index ].appendChild( element );
 		} );
 	}
+
+	applyDefaultStateCondition() {
+		if ( ! this.elements ) {
+			return;
+		}
+
+		const { $titles, $items, $accordionContent } = this.getDefaultElements(),
+			{ default_state: currentState } = this.getElementSettings(),
+			{ default_state: defaultState } = this.getDefaultSettings();
+
+		if ( currentState === defaultState ) {
+			this.open( $items[ 0 ], $titles[ 0 ], $accordionContent[ 0 ] );
+		} else {
+			$items.each( ( _, item ) => item.removeAttribute( 'open' ) );
+		}
+	}
+
+	bindEvents() {
+		this.bindAnimationListeners();
+	}
+
+	unbindEvents() {
+		this.removeAnimationListeners();
+	}
+
+	bindAnimationListeners() {
+		const { $titles, $items, $accordionContent } = this.getDefaultElements();
+
+		$titles.each( ( index, title ) => {
+			title.addEventListener( 'click', ( e ) => {
+				this.clickListener( e, $items, $accordionContent, index, title );
+			} );
+		} );
+	}
+
+	clickListener( e, $items, $accordionContent, index, title ) {
+		e.preventDefault();
+
+		const detailsNode = $items[ index ],
+			contentNode = $accordionContent[ index ];
+
+		if ( this.isClosing || ! $items[ index ].open ) {
+			this.open( detailsNode, title, contentNode );
+		} else if ( this.isExpanding || $items[ index ].open ) {
+			this.shrink( detailsNode, title );
+		}
+	}
+
+	shrink( el, summary ) {
+		el.style.overflow = 'hidden';
+
+		this.isClosing = true;
+
+		const startHeight = `${ el.offsetHeight }px`,
+			endHeight = `${ summary.offsetHeight }px`;
+
+		if ( this.animation ) {
+			this.animation.cancel();
+		}
+
+		this.animation = el.animate( {
+			height: [ startHeight, endHeight ],
+		}, {
+			duration: 1500,
+			easing: 'ease-out',
+		} );
+
+		this.animation.onfinish = () => this.onAnimationFinish( el, false );
+		this.animation.oncancel = () => this.isClosing = false;
+	}
+
+	open( el, summary, content ) {
+		el.style.overflow = 'hidden';
+		el.style.height = `${ el.offsetHeight }px`;
+		el.open = true;
+		window.requestAnimationFrame( () => this.expand( el, summary, content ) );
+	}
+
+	expand( el, summary, content ) {
+		this.isExpanding = true;
+
+		const fallbackHeightForEditor = 214,
+			contentHeight = content ? content.offsetHeight : fallbackHeightForEditor;
+
+		const startHeight = `${ el.offsetHeight }px`,
+			endHeight = `${ summary.offsetHeight + contentHeight }px`;
+
+		if ( this.animation ) {
+			this.animation.cancel();
+		}
+
+		this.animation = el.animate( {
+			height: [ startHeight, endHeight ],
+		}, {
+			duration: 1500,
+			easing: 'ease-out',
```

Comment:
Same as above.
As well, I see that open() and shrink() functions share some code.
Can we have this code in a single place?
Just in order to avoid code duplication.